agent/grpc-external/services/peerstream/stream_tracker.go

```diff
@@ -156,7 +156,7 @@ func (t *Tracker) IsHealthy(s Status) bool {
 	// If last Nack is after last Ack, it means the peer is unable to
 	// handle our replication message
 	if s.LastAck == nil {
```

Comment:
Are these if-statements even required now that we set the value to nil? Also, I think there will be a potential nil-pointer panic just a couple lines below due to this logic change.